src/main/java/rockset/AvroParser.java

```diff
@@ -56,7 +56,50 @@ private boolean isLogicalType(Schema schema) {
     return LOGICAL_TYPE_CONVERTERS.containsKey(schema.name());
   }
 
-  private Map<String, Object> convertLogicalTypes(Schema valueSchema, Map<String, Object> map) {
+  private boolean isScalarType(Schema schema) {
+    return (schema.type() == Type.STRING ||
+            schema.type() == Type.BYTES ||
+            schema.type() == Type.BOOLEAN ||
+            schema.type() == Type.INT8 ||
+            schema.type() == Type.INT16 ||
+            schema.type() == Type.INT32 ||
+            schema.type() == Type.INT64 ||
+            schema.type() == Type.FLOAT32 ||
+            schema.type() == Type.FLOAT64);
+  }
+
+  public List<Object> convertLogicalTypesArray(Schema schema, List<Object> arr) {
+    List<Object> res = new ArrayList<>();
+
+    for (Object e : arr) {
+      if (isLogicalType(schema.valueSchema())) {
```

Comment:
This and the below method handling maps look pretty identical. I think we can dedup by extracting a method `Object convertType(Schema, Object)` that handles all types - scalar, logical and other avro types. This might simplify the array and map methods to something like this:
```
arr:
for (Object e : arr) {
  res.add(convertType(schema, e));
}
map:
for (Entry e : map) {
  e.setValue(convertType(schema, e.getValue());
}
```
}

Reply:
👍